### ISSUE TITLE
[Question] What are the reasons for choosing the current Thompson sampling algorithm?

### ISSUE THREAD

**Issue by bperun123** [facebook/Ax#348]
Hey guys,

I am quite new to optimization and Thompson Sampling - so, sorry for this "stupid" question.

You describe the algorithm of thompson sampling quite well in a high-level view on your website (https://ax.dev/docs/banditopt.html):
`
Ax relies on the simple and effective Thompson sampling algorithm for performing bandit optimization. There is a clear intuition to this method: select a parameterization (referred to from now on as an "arm") with a probability proportional to that arm being the best. This algorithm is easy to implement and has strong guarantees of converging to an arm that is close to the best — all without any human intervention. To understand how this works, we describe an advertising optimization problem in which we want to choose arms which maximize the click-through rate (CTR) and the rewards are binary: either clicks (successes) or views without clicks (failures).

As we run the experiment, we develop more precise estimates of the performance of each arm. More precisely, in each iteration, we draw samples from the distribution of plausible effects for each arm, and we record the largest sampled value. We repeat this process many times, until settling on a final distribution of maximal arms, which determines how we assign users to arms going forward. This process rapidly narrows down our set of arms to only the very best performers.
`

My questions:

1. Why have you used a more advanced Thompson Sampling algorithm in comparison to a simple beta distribution Thompson Sampling for AX (something like that: https://hackernoon.com/use-beta-distribution-and-thompson-sampling-to-beat-the-multi-armed-bandit-at-the-casino-ulgp3272)? Has it something to do with the "kind of optimization"? What were the reasons to choose it?

2. Is there a paper to understand the current Thompson Sampling algorithm of AX in more detail from a mathematical point of view?

Thanks a lot in advance!

**Comment by eytan** [comment]
We found that empirically, the normal approximation works better in practice than the beta for binary outcomes, even though the beta more correctly captures the uncertainty. Ax uses an empirical Bayes estimator which performs even better than the normal. See our KDD 2019 paper for more details   https://arxiv.org/pdf/1904.12918.pdf

I don’t think we show how normal works better than beta in the KDD paper, but we applied a similar methodology where we took data from a diverse set of marketing campaigns that had click throughs on both smaller (<1%) and larger (~10%), and the normal approximation yielded superior results, perhaps because it is slightly more exploitative.

Hope this helps!
